### PR TITLE
Move sandboxes into repository

### PR DESCRIPTION
**Pull request by ChristianMurphy**
support both stackblitz and codesandbox

### Initial checklist

*   [x] I read the support docs 
*   [x] I read the contributing guide 
*   [x] I agree to follow the code of conduct 
*   [x] I searched issues and couldn’t find anything (or linked relevant results below) 
*   [x] If applicable, I’ve added docs and tests

### Description of changes

* brings sandbox example into repository so they can be more easily managed by community
* updates parcel to version 2
* support both stackblitz and codesandbox as host
* improved commentary on updating dependencies in sandbox

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/1-bug.yml [hunk 1]
```diff
@@ -40,9 +40,12 @@ body:
...
 
-        Some CodeSandbox starters:
-        - [rehype](https://codesandbox.io/s/rehype-debug-po7c5) (for html to html, with parse5)
-        - [rehype-dom](https://codesandbox.io/s/rehype-dom-debug-vqcgs) (for html to html, with browser DOM)
+        Some starters:
+
+        | description | codesandbox | stackblitz |
+        | - | - | - |
+        | rehype (for html to html, with parse5) | [codesandbox](https://codesandbox.io/s/github/rehypejs/.github/tree/main/sandbox-templates/rehype-debug) | [stackblitz](https://stackblitz.com/github/rehypejs/.github/tree/main/sandbox-templates/rehype-debug) |
+        | rehype-dom (for html to html, with browser DOM) | [codesandbox](https://codesandbox.io/s/github/rehypejs/.github/tree/main/sandbox-templates/rehype-dom-debug) | [stackblitz](https://stackblitz.com/github/rehypejs/.github/tree/main/sandbox-templates/rehype-dom-debug) |
```
Comment (ChristianMurphy) [inline .github/ISSUE_TEMPLATE/1-bug.yml:48]: sample links while PR before PR is merged

| description | codesandbox | stackblitz |
| - | - | - |
| rehype (for html to html, with parse5) | [codesandbox](https://codesandbox.io/s/github/ChristianMurphy/.github-rehype/tree/sandbox-in-repo/sandbox-templates/rehype-debug) | [stackblitz](https://stackblitz.com/github/ChristianMurphy/.github-rehype/tree/sandbox-in-repo/sandbox-templates/rehype-debug) |
| rehype-dom (for html to html, with browser DOM) | [codesandbox](https://codesandbox.io/s/github/ChristianMurphy/.github-rehype/tree/sandbox-in-repo/sandbox-templates/rehype-dom-debug) | [stackblitz](https://stackblitz.com/github/ChristianMurphy/.github-rehype/tree/sandbox-in-repo/sandbox-templates/rehype-dom-debug) |

File: sandbox-templates/rehype-debug/index.html [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
```
Comment (wooorm) [inline sandbox-templates/rehype-debug/index.html]:
```suggestion
<!doctype html>
```

File: sandbox-templates/rehype-debug/index.html [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+<html>
+  <head>
+    <title>rehype sandbox</title>
+    <meta charset="UTF-8" />
```
Comment (wooorm) [inline sandbox-templates/rehype-debug/index.html]:
```suggestion
    <meta charset="utf-8">
```

File: sandbox-templates/rehype-debug/index.html [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
+<html>
```
Comment (wooorm) [inline sandbox-templates/rehype-debug/index.html]:
```suggestion
<html lang="en">
```

File: sandbox-templates/rehype-debug/src/index.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+const sourceHtml = `
+<html>
+  <head></head>
+  <body>
+    <h1>title</h1>
+    <p>lorem ipsum</p>
+  </body>
+</html>
+`;
```
Comment (wooorm) [inline sandbox-templates/rehype-debug/src/index.js]: This document is missing a doctype and a title element for it to be a conforming HTML document.

File: sandbox-templates/rehype-debug/src/index.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+// ⚠️ Important! Please make sure the dependencies are up to date.
+// On code sandbox, you can refresh them in the Dependencies section (left-bottom)
+// On stackblitz, you can open the package.json file, update the versions,
+// then run npm install in the stackblitz terminal
+
+import { rehype } from "rehype";
```
Comment (wooorm) [inline sandbox-templates/rehype-debug/src/index.js]: These files should be formatted. Example: https://github.com/remarkjs/.github/blob/1c6b1aa982f9de3742e09786ec0b6256c838b707/package.json#L27-L61

File: sandbox-templates/rehype-debug/index.html [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+  <head>
+    <title>rehype sandbox</title>
+    <meta charset="UTF-8" />
+  </head>
+
+  <body>
```
Comment (wooorm) [inline sandbox-templates/rehype-debug/index.html]:
```suggestion
  </head>
  <body>
```

File: sandbox-templates/rehype-debug/index.html [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    <h2>source</h2>
+    <pre id="source"></pre>
+    <h2>result</h2>
+    <pre id="result"></pre>
+    <h2>error</h2>
+    <pre id="error">none</pre>
+
+    <script type="module" src="src/index.js"></script>
```
Comment (wooorm) [inline sandbox-templates/rehype-debug/index.html]:
```suggestion
    <pre id="error">none</pre>
    <script type="module" src="src/index.js"></script>
```

File: sandbox-templates/rehype-dom-debug/index.html [hunk 9]
```diff
@@ -0,0 +1,18 @@
+<!DOCTYPE html>
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/index.html]:
```suggestion
<!doctype html>
```

File: sandbox-templates/rehype-dom-debug/index.html [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+<html>
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/index.html]:
```suggestion
<html lang="en">
```

File: sandbox-templates/rehype-dom-debug/index.html [hunk 11]
```diff
@@ -0,0 +1,18 @@
...
+  <head>
+    <title>rehype-dom sandbox</title>
+    <meta charset="UTF-8" />
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/index.html]:
```suggestion
    <meta charset="utf-8">
```

File: sandbox-templates/rehype-dom-debug/index.html [hunk 12]
```diff
@@ -0,0 +1,18 @@
...
+  </head>
+
+  <body>
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/index.html]:
```suggestion
  </head>
  <body>
```

File: sandbox-templates/rehype-dom-debug/index.html [hunk 13]
```diff
@@ -0,0 +1,18 @@
...
+    <h2>source</h2>
+    <pre id="source"></pre>
+    <h2>result</h2>
+    <pre id="result"></pre>
+    <h2>error</h2>
+    <pre id="error">none</pre>
+
+    <script type="module" src="src/index.js"></script>
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/index.html]:
```suggestion
    <pre id="error">none</pre>
    <script type="module" src="src/index.js"></script>
```

File: sandbox-templates/rehype-dom-debug/src/index.js [hunk 14]
```diff
@@ -0,0 +1,27 @@
+// ⚠️ Important! Please make sure the dependencies are up to date.
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/src/index.js:1]: This file should be formatted (see above)

File: sandbox-templates/rehype-dom-debug/src/index.js [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+import { rehypeDom } from "rehype-dom";
+
+const sourceHtml = `
+<html>
+  <head></head>
+  <body>
+    <h1>title</h1>
+    <p>lorem ipsum</p>
+  </body>
+</html>
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/src/index.js:19]: doctype and title

File: sandbox-templates/rehype-dom-debug/src/index.js [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+document.getElementById("source").textContent = sourceHtml;
+
+rehypeDom()
+  .use({ settings: { fragment: false } })
+  // add any plugins here
+  .process(sourceHtml)
+  .then(
+    (file) => (document.getElementById("result").textContent = String(file))
+  )
+  .catch((err) => (document.getElementById("error").innerHTML = err));
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/src/index.js]: async/await and `main` with a `.catch`.

File: .gitignore [hunk 17]
```diff
@@ -1 +1,4 @@
 node_modules/
+.parcel-cache/
+dist/
+package-lock.json
```
Comment (wooorm) [inline .gitignore:4]: Good to put these Parcel-specific patterns inside the sandboxes, where theu are needed. That way, they are included in forked sandboxes.

File: sandbox-templates/rehype-debug/index.html (hunk 2 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-debug/index.html]: updated

File: sandbox-templates/rehype-debug/index.html (hunk 3 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-debug/index.html]: updated

File: sandbox-templates/rehype-debug/index.html (hunk 4 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-debug/index.html]: updated

File: sandbox-templates/rehype-debug/src/index.js (hunk 5 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-debug/src/index.js]: updated

File: sandbox-templates/rehype-debug/src/index.js (hunk 6 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-debug/src/index.js]: updated

File: sandbox-templates/rehype-dom-debug/index.html (hunk 9 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-dom-debug/index.html]: updated

File: sandbox-templates/rehype-dom-debug/index.html (hunk 11 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-dom-debug/index.html]: updated

File: sandbox-templates/rehype-dom-debug/src/index.js (hunk 14 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-dom-debug/src/index.js:1]: updated

File: sandbox-templates/rehype-dom-debug/src/index.js (hunk 16 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-dom-debug/src/index.js]: updated

File: sandbox-templates/rehype-dom-debug/index.html (hunk 10 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-dom-debug/index.html]: updated

File: sandbox-templates/rehype-dom-debug/index.html (hunk 13 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-dom-debug/index.html]: updated

File: sandbox-templates/rehype-dom-debug/index.html (hunk 12 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-dom-debug/index.html]: updated

File: sandbox-templates/rehype-dom-debug/src/index.js (hunk 15 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-dom-debug/src/index.js:19]: updated

File: sandbox-templates/rehype-debug/index.html (hunk 8 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-debug/index.html]: updated

File: sandbox-templates/rehype-debug/index.html (hunk 7 above)
Comment (ChristianMurphy) [reply sandbox-templates/rehype-debug/index.html]: updated

File: .gitignore (hunk 17 above)
Comment (ChristianMurphy) [reply .gitignore:4]: I don't think the sandboxes are themselves git repos when used 🤔 
but I can add a gitignore to each anyway 🤷‍♂️

File: .github/pull-request-template.md [hunk 18]
```diff
@@ -8,11 +8,11 @@
...
-*   [ ] I read the support docs <!-- https://github.com/rehypejs/.github/blob/main/support.md -->
-*   [ ] I read the contributing guide <!-- https://github.com/rehypejs/.github/blob/main/contributing.md -->
-*   [ ] I agree to follow the code of conduct <!-- https://github.com/rehypejs/.github/blob/main/code-of-conduct.md -->
-*   [ ] I searched issues and couldn’t find anything (or linked relevant results below) <!-- https://github.com/search?q=user%3Arehypejs&type=Issues -->
-*   [ ] If applicable, I’ve added docs and tests
+- [ ] I read the support docs <!-- https://github.com/rehypejs/.github/blob/main/support.md -->
+- [ ] I read the contributing guide <!-- https://github.com/rehypejs/.github/blob/main/contributing.md -->
+- [ ] I agree to follow the code of conduct <!-- https://github.com/rehypejs/.github/blob/main/code-of-conduct.md -->
+- [ ] I searched issues and couldn’t find anything (or linked relevant results below) <!-- https://github.com/search?q=user%3Arehypejs&type=Issues -->
+- [ ] If applicable, I’ve added docs and tests
```
Comment (wooorm) [inline .github/pull-request-template.md]: `npm run format` should revert this change?

File: sandbox-templates/rehype-debug/src/index.js [hunk 19]
```diff
@@ -0,0 +1,34 @@
...
+  rehype()
+    // Add any plugins here
+    .process(sourceHtml)
+    .then((file) => {
+      document.querySelector('#result').textContent = String(file)
+    })
+}
+
+main().catch((error) => {
+  document.querySelector('#error').innerHTML = error
```
Comment (wooorm) [inline sandbox-templates/rehype-debug/src/index.js]:
```suggestion
  document.querySelector('#error').textContent = error
```

File: sandbox-templates/rehype-debug/src/index.js [hunk 20]
```diff
@@ -0,0 +1,34 @@
...
+// On stackblitz, you can open the package.json file, update the versions,
+// then run npm install in the stackblitz terminal
+
+import {rehype} from 'rehype'
+
+const sourceHtml = `
+<!doctype html>
+<html>
+  <head>
+    <title>sandbox document</title>
```
Comment (wooorm) [inline sandbox-templates/rehype-debug/src/index.js:13]:
```suggestion
    <meta charset="utf-8">
    <title>sandbox document</title>
```

File: sandbox-templates/rehype-dom-debug/index.html [hunk 21]
```diff
@@ -0,0 +1,16 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <title>rehype-dom sandbox</title>
+    <meta charset="utf-8" />
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/index.html]:
```suggestion
    <meta charset="utf-8">
    <title>rehype-dom sandbox</title>
```

File: sandbox-templates/rehype-debug/index.html [hunk 22]
```diff
@@ -0,0 +1,16 @@
+<!doctype html>
+<html lang="en">
+  <head>
+    <title>rehype sandbox</title>
+    <meta charset="utf-8" />
```
Comment (wooorm) [inline sandbox-templates/rehype-debug/index.html]:
```suggestion
    <meta charset="utf-8">
    <title>rehype sandbox</title>
```

File: sandbox-templates/rehype-dom-debug/src/index.js [hunk 23]
```diff
@@ -0,0 +1,35 @@
...
+// On stackblitz, you can open the package.json file, update the versions,
+// then run npm install in the stackblitz terminal
+
+import {rehypeDom} from 'rehype-dom'
+
+const sourceHtml = `
+<!doctype html>
+<html>
+  <head>
+    <title>sandbox document</title>
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/src/index.js:13]:
```suggestion
    <meta charset="utf8">
    <title>sandbox document</title>
```

File: sandbox-templates/rehype-dom-debug/src/index.js [hunk 24]
```diff
@@ -0,0 +1,35 @@
...
+    .use({settings: {fragment: false}})
+    // Add any plugins here
+    .process(sourceHtml)
+    .then((file) => {
+      document.querySelector('#result').textContent = String(file)
+    })
+}
+
+main().catch((error) => {
+  document.querySelector('#error').innerHTML = error
```
Comment (wooorm) [inline sandbox-templates/rehype-dom-debug/src/index.js]:
```suggestion
  document.querySelector('#error').textContent = error
```

**Review by wooorm — COMMENTED** [review]
Few nits, then it’s good!

File: .github/pull-request-template.md (hunk 18 above)
Comment (ChristianMurphy) [reply .github/pull-request-template.md]: prettier changed this before I added a .prettierignore, remark appears to be ignoring it

Comment (ChristianMurphy) [reply .github/pull-request-template.md]: manually fixed

**Review by wooorm — APPROVED** [review]

**Comment by wooorm** [comment]
sweeet, thanks!
